pandas/tests/groupby/test_function.py

```diff
@@ -1087,3 +1087,11 @@ def test_groupby_sum_below_mincount_nullable_integer():
     result = grouped.sum(min_count=2)
     expected = DataFrame({"b": [pd.NA] * 3, "c": [pd.NA] * 3}, dtype="Int64", index=idx)
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_index_object_dtype():
```

Comment:
I think `pandas/tests/groupby/aggregate/test_aggregate.py` might be a better place for this test - looks like there are some pretty similar ones

Reply:
Done!